build/configure/app-configure-step.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+var utils = require('utils');
+
+var DependencyGraph = require('./dependency-graph');
+var BuildConfig = require('./build-config');
+
+// Default building steps before app's default building steps.
+// TODO: We should have a module to map these steps with different app's own
+//       config data, so that app could easily to add/remove/reorder steps for
+//       different situations.
+const POST_APP_BUILD_STEPS = [
+  'media-resolution',
+  'post-manifest',
+  'multilocale',
+  'copy-build-stage-data',
+  'webapp-optimize',
+  'webapp-zip'
+];
+
+/**
+ * AppConfigureStep would generate all the data for build back-end, ideally it
+ * should only manage the steps relating to app itself, like app's own
+ * buildscript and POST_APP_BUILD_STEPS.
+ *
+ * @param {object} options - build_config from main Makefile.
+ * @param {string} appDir - the app's source dir path.
+ * @constructor
+ */
+var AppConfigureStep = function(options, appDir) {
+  this.appDir = appDir;
+  this.webapp = utils.getWebapp(appDir, options);
+  this.name = this.webapp.sourceDirectoryName;
+  this.options = options;
+};
+
+AppConfigureStep.prototype = {
+  /**
+   * Configure start from here.
+   *
+   * @param {[string]} previousSteps - the steps we should finish before app's
```

Comment:
If you would like to describe API in jsdoc format, @param should be {string|string[]} for single string or multiple strings.
See also http://usejsdoc.org/tags-param.html#multiple-types-and-repeatable-parameters